zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,970 @@
+fn copy_integer_bytes_le<F: FieldExt>(
+    region: &mut Region<'_, F>,
+    name: &str,
+    src: &[AssignedValue<F>; 32],
+    dst: &[Column<Advice>; 32],
+    offset: usize,
+) -> Result<(), Error> {
+    for (i, byte) in src.iter().enumerate() {
+        let assigned_cell = region.assign_advice(
+            || format!("{} byte {}", name, i),
+            dst[i],
+            offset,
+            || byte.value().ok_or(Error::Synthesis),
+        )?;
+        region.constrain_equal(assigned_cell.cell(), byte.cell())?;
+    }
+    Ok(())
+}
+
+#[derive(Debug, Clone)]
+pub struct SignVerifyConfig<F: FieldExt> {
+    q_enable: Selector,
+    pk_hash: [Column<Advice>; 32],
+    // When address is 0, we disable the signature verification by using a dummy pk, msg_hash and
+    // signature which is not constrainted to match msg_hash_rlc nor the address.
+    address: Column<Advice>,
+    address_is_zero: IsZeroConfig<F>,
+    address_inv: Column<Advice>,
+    msg_hash_rlc: Column<Advice>,
+
+    // ECDSA
+    main_gate_config: MainGateConfig,
+    range_config: RangeConfig,
+    // First 32 cells are coord x in little endian, following 32 cells are coord y in little
+    // endian.
+    pk: [[Column<Advice>; 32]; 2],
+    msg_hash: [Column<Advice>; 32],
+    power_of_randomness: [Expression<F>; POW_RAND_SIZE],
+
+    // [is_enabled, input_rlc, input_len, output_rlc]
+    keccak_table: [Column<Advice>; 4],
+}
+
+impl<F: FieldExt> SignVerifyConfig<F> {
+    pub fn new(
+        meta: &mut ConstraintSystem<F>,
+        power_of_randomness: [Expression<F>; POW_RAND_SIZE],
+    ) -> Self {
+        let q_enable = meta.complex_selector();
+
+        let pk = [(); 2].map(|_| [(); 32].map(|_| meta.advice_column()));
+        pk.iter()
+            .for_each(|coord| coord.iter().for_each(|c| meta.enable_equality(*c)));
+
+        let msg_hash = [(); 32].map(|_| meta.advice_column());
+        msg_hash.iter().for_each(|c| meta.enable_equality(*c));
+
+        let address = meta.advice_column();
+        meta.enable_equality(address);
+
+        let pk_hash = [(); 32].map(|_| meta.advice_column());
+
+        let msg_hash_rlc = meta.advice_column();
+        meta.enable_equality(msg_hash_rlc);
+
+        let address_inv = meta.advice_column();
+        let address_is_zero = IsZeroChip::configure(
+            meta,
+            |meta| meta.query_selector(q_enable),
+            |meta| meta.query_advice(address, Rotation::cur()),
+            address_inv,
+        );
+        // is_not_padding == address != 0
+        let is_not_padding = not::expr(address_is_zero.is_zero_expression.clone());
+
+        // lookup keccak table
+        let keccak_table = [(); 4].map(|_| meta.advice_column());
+
+        // Ref. spec SignVerifyChip 1. Verify that keccak(pub_key_bytes) = pub_key_hash
+        // by keccak table lookup, where pub_key_bytes is built from the pub_key
+        // in the ecdsa_chip
+        // keccak lookup
+        meta.lookup_any("keccak", |meta| {
```

Comment:
I thought the same! ahhahah

That's why I wanted to meet with you. Since we can do a proxy-table to simplify the API (at a cost obviously). And I thought that once the state_tag stuff is handled (I'm working on it), We should meet with @miha-stopar and @ed255 to see what are they expecting exactly as API.